src/main/java/mekanism/common/block/property/PropertyConnection.java

```diff
@@ -5,7 +5,7 @@
 import mekanism.common.tile.transmitter.TileEntitySidedPipe.ConnectionType;
 import net.minecraftforge.common.property.IUnlistedProperty;
 
-public class PropertyConnection implements IUnlistedProperty<PropertyConnection>
+public final class PropertyConnection implements IUnlistedProperty<PropertyConnection>
```

Comment:
Why `final`?

Reply:
Looking more closely, I guess it's not strictly necessary. I can take it out if you want.

Reply:
If you have a god reason for it we can do it, but finality for finality's sake is just no.

Reply:
My reasoning is that PropertyConnection is an IUnlistedProperty of itself, which is unusual. It's more common to see, for example, PropertyFloat being an IUnlistedProperty of Float. If someone were to subclass this, they wouldn't be able to make getType() return the right thing anymore, since it's defined to return a `Class<V>`, not a `Class<? extends V>`.

Reply:
Not really unusual, why have 2 classes, when one will do.

`getType` should return the subclass' `Class`, and generics don't matter at runtime

Reply:
getType can't return the subclass's Class, though, because the subclass would still be an IUnlistedProperty<PropertyConnection>, so it would still have to return a Class<PropertyConnection>, which a subclass's class isn't. And although generics are erased at runtime, doing unsafe things could still result in a ClassCastException at runtime.

Reply:
well, actually having looked further, were this only a container class (ie not a property of itself) that would make sense.

It currently has no methods, just properties, but you could subclass it to have some and still put it in the same `IUnlistedProperty` storage in the blockstate, no?

Reply:
I guess so. I'll remove `final` for now.